src/test/regress/sql/multi_task_assignment_policy.sql

```diff
@@ -212,7 +206,35 @@ INSERT INTO explain_outputs
 -- given that we're in the same transaction, the count should be 2
 -- since there are two different worker nodes
 SELECT count(DISTINCT value) FROM explain_outputs;
+TRUNCATE explain_outputs;
 
+-- test that the round robin policy detects the anchor shard correctly
```

Comment:
Might be worth adding something like "we should not pick a reference table shard as the anchor shard when joining with a distributed table".

Reply:
I added this explanation and revised my other comments in this test file. Thanks for the advice